piperci_gman/resource.py

```diff
@@ -19,34 +19,100 @@ def InternalError(self, message=None):
         message = f': {message}' if message else ''
         return ({'message': f'Internal Server Error{message}'}, 500)
 
-    def get_task_states(self, events):
+    def task_states(self, events):
         '''Parses task events to identify which state the task is in.'''
 
-        running = ('started', 'received')
+        running = ('started', 'received',)
         completed = ('completed',)
         failed = ('failed',)
+        pending = ('delegated',)
 
         states = {'running': [],
                   'completed': [],
-                  'failed': []}
+                  'failed': [],
+                  'pending': [],
+                  'received': []}
+
+        pending_tasks = []
```

Comment:
More of a question...
What is the point of setting pending_tasks here? Why not just use states['pending'] ?

Reply:
hanging thing from dev, good catch